fuel-merkle/src/sparse/in_memory.rs

```diff
@@ -32,6 +34,91 @@ impl MerkleTree {
         }
     }
 
+    pub fn from_set<I, D>(set: I) -> Self
+    where
+        I: Iterator<Item = (Bytes32, D)>,
+        D: AsRef<[u8]>,
+    {
+        let tree = SparseMerkleTree::from_set(Storage::new(), set).expect("`Storage` can't return error");
+        Self { tree }
+    }
+
+    pub fn root_from_set<I, D>(set: I) -> Bytes32
+    where
+        I: Iterator<Item = (Bytes32, D)>,
+        D: AsRef<[u8]>,
+    {
+        #[derive(Default)]
+        struct EmptyStorage;
+
+        impl StorageInspect<NodesTable> for EmptyStorage {
+            type Error = core::convert::Infallible;
+
+            fn get(&self, _: &Bytes32) -> Result<Option<Cow<Primitive>>, Self::Error> {
+                Ok(None)
+            }
+
+            fn contains_key(&self, _: &Bytes32) -> Result<bool, Self::Error> {
+                Ok(false)
+            }
+        }
+
+        impl StorageMutate<NodesTable> for EmptyStorage {
+            fn insert(&mut self, _: &Bytes32, _: &Primitive) -> Result<Option<Primitive>, Self::Error> {
+                Ok(None)
+            }
+
+            fn remove(&mut self, _: &Bytes32) -> Result<Option<Primitive>, Self::Error> {
+                Ok(None)
+            }
+        }
+
+        let tree = sparse::MerkleTree::<NodesTable, _>::from_set(EmptyStorage::default(), set)
+            .expect("`Storage` can't return error");
+        tree.root()
+    }
+
+    pub fn nodes_from_set<I, D>(set: I) -> (Bytes32, Vec<(Bytes32, Primitive)>)
```

Comment:
Other than in the benchmarks, what is this used for?

Reply:
The `init_contract_state` method uses it in the [PR](https://github.com/FuelLabs/fuel-core/pull/1207/files#diff-aae50a9fdae7f687d8bf406ffb86fd85b200bb01aeff2575709a015fc3705660)